csharp/Ice/Filesystem/Server/MDirectory.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) ZeroC, Inc.
+
+namespace Server;
+
+/// <summary>Provides an in-memory implementation of the Slice interface Directory.</summary>
+internal class MDirectory : Filesystem.DirectoryDisp_
+{
+    private readonly string _name;
+    private readonly List<Filesystem.NodePrx> _contents = new();
+
+    /// <inheritdoc/>
+    public override string Name(Ice.Current current) => _name;
```

Comment:
Unlike the C++ version, we can't define a MNode and reuse it. That's because XxxDisp_ is a class.